src/client.rs

```diff
@@ -58,9 +58,7 @@ impl Default for Options {
 
 /// Client represents an object that can create new builders and events and send them
 /// somewhere.
-#[derive(Debug, Clone)]
```

Comment:
I've start going through a closer look and this is probably going to be the biggest change/pain point for consumers of the library.

I experimented with a library `beeline-rust` which uses this one, providing a `actix-web` middleware and it requires really drastic changes to cope with this `Clone` disappearing.

Reply:
for `opentelemetry-honeycomb`, I was able to solve this by wrapping the client in an `Arc+RwLock`: https://github.com/ramosbugs/opentelemetry-honeycomb-rs/blob/ca82ecd99bdfcf9357547d6079e31d70a5bfb9b4/src/lib.rs#L168-L176

then sending events just involves acquiring the read lock: https://github.com/ramosbugs/opentelemetry-honeycomb-rs/blob/ca82ecd99bdfcf9357547d6079e31d70a5bfb9b4/src/lib.rs#L332-L335

would that solution work for `actix-web`, or is there something about actix that makes it more difficult?

Reply:
I think it would. I'm going to play with it tomorrow am and come back to this. I'm still quite keen on merging this PR.